### ISSUE TITLE
All dependencies are archived and deprecated. Let's upgrade

### ISSUE THREAD

**Issue by LefterisJP** [filips123/ContentHashPy#1]
The `py-cid` dependency is unmaintained https://github.com/ipld/py-cid and archived.

This means that anyone trying to use this repo in a big project will end up hitting constraints due to the version restrictions. For example we at rotki just tried here: https://github.com/rotki/rotki/pull/6259

And we hit this:

```
The conflict is caused by:
    scalecodec 1.2.2 depends on base58>=2.0.1
    substrate-interface 1.6.2 depends on base58<3 and >=1.0.3
    py-cid 0.3.0 depends on base58<2.0 and >=1.0.2
```

Looking here https://github.com/ipld/ipld/issues/251 it seems that the new package to use is: https://github.com/pinnaculum/py-multiformats-cid ?

Would it be possible to update and bump the version?

**Comment by LefterisJP** [comment]
Hmm hit the same issue with multihash and the other dependency. Basically all dependencies are archived :cry: 

Will try to update/rewrite what I can to make it work for us.

**Comment by filips123** [comment]
I released 2.0.0 with changes from #3. I've also reformated the code, added type hints and changed a minimum Python version to 3.8.
